File: 49-group-anagrams/group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        vector<vector<string>>ans;
        unordered_map<string, vector<string>>cmap;
        for(int i = 0;i<strs.size();i++){
            string strsval = strs[i];
            sort(strsval.begin(), strsval.end());
            cmap[strsval].push_back(strs[i]);
        }
        for(auto i : cmap){
            ans.push_back(i.second);
        }
        return ans;
    }
};

File: 49-group-anagrams/group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        res = defaultdict(list)

        for s1 in strs:
            count = [0] * 26
            for ch in s1:
                count[ord(ch) - ord('a')] += 1
            res[tuple(count)].append(s1)
        
        return list(res.values())
        